### ISSUE TITLE
bug(pin-versions-in-pip): false positive when using pipx DL3013

### ISSUE THREAD

**Issue by blyedev** [hadolint/hadolint#1067]
- [x] This is a bug report
- [ ] This is a feature request
- [x] I searched existing issues before opening this one

### Expected behavior

The following is part of my Dockerfile
```Dockerfile
ENV POETRY_VERSION=1.8.5
RUN pipx install --python "$(which python)" "poetry==$POETRY_VERSION" 
```

It should not trigger any linting fails

### Actual behavior

The following warning is issued
```
1. hadolint: Pin versions in pip. Instead of `pip install <package>` use `pip install <package>==<version>` or `pip install --requirement <requirements file>` [DL3013]
```

### Steps to reproduce the behavior

**Output of `hadolint --version` or
  `docker run --rm hadolint/hadolint hadolint --version` or
  `docker run --rm ghcr.io/hadolint/hadolint hadolint --version`:**

I'm using mason in neovim so this is the only output i have.
```bash
    ◍ hadolint
      Dockerfile linter, validate inline bash, written in Haskell.

      installed version v2.12.0                              
      homepage          https://github.com/hadolint/hadolint 
      languages         Docker                               
      categories        Linter                               
      executables       hadolint                             

```

**Additional environment details (OS, stack version, etc.)**
I'm using Arch btw. Neovim too
